staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/lastappliedupdater.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type lastAppliedUpdater struct {
+	fieldManager Manager
+}
+
+var _ Manager = &lastAppliedUpdater{}
+
+func NewLastAppliedUpdater(fieldManager Manager) Manager {
+	return &lastAppliedUpdater{
+		fieldManager: fieldManager,
+	}
+}
+
+// Update implements Manager.
+func (f *lastAppliedUpdater) Update(liveObj, newObj runtime.Object, managed Managed, manager string) (runtime.Object, Managed, error) {
+	return f.fieldManager.Update(liveObj, newObj, managed, manager)
+}
+
+// Apply keeps the client-side apply annotation up to date with
+// server-side apply managed fields
+func (f *lastAppliedUpdater) Apply(liveObj runtime.Object, newObj runtime.Object, managed Managed, manager string, force bool) (runtime.Object, Managed, error) {
+	err := f.updateLastApplied(newObj)
```

Comment:
Oops, I think we need to serialize `newObj` as the annotation, but we need to set the annotation on `liveObj` to make this work

Reply:
Correct!

Reply:
Okay, fixed. Will need an e2e or integration test for this.

Tested on a local cluster:

```
$ cat <<EOF > test.yaml
→ apiVersion: v1
→ kind: Pod
→ metadata:
→   namespace: default
→   name: pause
→ spec:
→   containers:
→   - image: k8s.gcr.io/pause:3.1
→     name: pause
→ EOF

$ ./cluster/kubectl.sh apply -f test.yaml
pod/pause created

$ ./cluster/kubectl.sh apply view-last-applied -f test.yaml
apiVersion: v1
kind: Pod
metadata:
  annotations: {}
  name: pause
  namespace: default
spec:
  containers:
  - image: k8s.gcr.io/pause:3.1
    name: pause

$ sed s/pause:3.1/pause:latest/g test.yaml -i

$ ./cluster/kubectl.sh apply -f test.yaml --server-side --force-conflicts
pod/pause serverside-applied

$ ./cluster/kubectl.sh apply view-last-applied -f test.yaml
apiVersion: v1
kind: Pod
metadata:
  name: pause
  namespace: default
spec:
  containers:
  - image: k8s.gcr.io/pause:latest
    name: pause

```